src/hugr/validate/test.rs

```diff
@@ -750,3 +750,51 @@ fn invalid_types() {
         SignatureError::TypeArgMismatch(TypeArgError::WrongNumberArgs(2, 1))
     );
 }
+
+#[test]
+fn parent_io_mismatch() {
+    let mut hugr = Hugr::new(NodeType::new_pure(ops::DFG {
+        signature: FunctionType::new(type_row![USIZE_T], type_row![USIZE_T]),
```

Comment:
The problem here is that the FunctionType has zero delta? Whereas if you did `.with_extension_delta(...XB_T...)` it'd pass? (Might be worth doing both ways via a loop / helper fn just to show the difference between working and not working)

Reply:
Added a comment explaining this